handlers/github_api_handler.js

```diff
@@ -0,0 +1,141 @@
+   * and then converting it back into an array. (Since sets can't contain duplicates)
+   * @param {Array} repoSet - A set of objects containing repository information
+   * @returns {Array} A unique set of objects containing repository information
+   */
+  removeDuplicates: function (repoSet) {
+    return _.uniqWith(repoSet, _.isEqual)
+  },
+  /**
+   * Gets a repository on github by its id and checks if the owner
+   * of the repo is the current user
+   * @param {Object} github - Allows for API calls to be made to the github api server
+   * @param {Object} repo_information - The returned object containing information on repositories contributed to by the user
+   * @param {String} user - The owner of the account to be validated
+   * @returns {Promise} Either returns that the repository is owned by the user or not
+   */
+  getRepoByID: function (github, repo_information, user) {
+    return new Promise((resolve, reject) => {
+      github.repos.getById({
+        id: repo_information.repo.id
+      }, (err, result) => {
+        if (err) {
+          throw err
+        }
+        if (result.owner.login !== user) {
+          resolve({
+            id: result.id,
+            isPublic: true
+          })
+        } else {
+          resolve({
+            id: result.id,
+            isPublic: false
+          })
+        }
+      })
+    })
+  },
+  /**
+   * Gets a page of GitHub account events, each page containing 100 events, which can be
+   * push events, pull request events, issue events etc...
+   * @param {Object} github - Allows for API calls to be made to the github api server
+   * @param {String} user - The owner of the account to be validated
+   * @param {Number} pageNumber - The page number of account events
+   * @param {Number} sixMonthsAgo - The current time minus 6 months
+   * @returns {Promise} Returns the number of repos that are not owned by the user and are unique
+   */
+  getPageOfRepos: function (github, user, pageNumber, sixMonthsAgo) {
+    return new Promise((resolve, reject) => {
+      let pset = []
+      github.activity.getEventsForUserPublic({
+        user,
+        page: pageNumber,
+        per_page: 100
+      }, (err, result) => {
+        if (err) {
+          throw err
+        }
+        let repoEvent = ''
+        result.forEach((event) => {
+          repoEvent = new Date(event.created_at)
+          if (sixMonthsAgo.getTime() < repoEvent.getTime()) {
+            if (event.type === 'PullRequestEvent') {
+              pset.push(githubAPI.getRepoByID(github, event, user))
+            }
+          }
+        })
+        Promise.all(pset).then((values) => {
+          let numRepos = 0
+          values = githubAPI.removeDuplicates(values)
+          values.forEach((element) => {
+            if (element.isPublic) {
+              numRepos++
+            }
+          })
+          resolve({
+            total: numRepos
+          })
```

Comment:
For the removeDuplicates, I used _.uniqWith(). I plan to use reduce to replace the forEach